core/src/types/operator/registry.rs

```diff
@@ -0,0 +1,242 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::cell::LazyCell;
+use std::collections::HashMap;
+
+use crate::services::*;
+use crate::*;
+
+// TODO: thread local or use LazyLock instead? this way the access is lock-free
+// TODO: should we expose the `GLOBAL_OPERATOR_REGISTRY` as public API at `crate::types::operator::GLOBAL_OPERATOR_REGISTRY`?
+thread_local! {
+    pub static GLOBAL_OPERATOR_REGISTRY: LazyCell<OperatorRegistry> = LazyCell::new(|| OperatorRegistry::with_enabled_services());
```

Comment:
MSRV check fails due to the usage of `LazyCell`. Should we update the MSRV or use another thing instead? 

I see this `TODO` about the `once_cell` crate usage: https://github.com/apache/opendal/blob/52c96bb8e8cb4d024ccab1f415c4756447c726dd/core/Cargo.toml#L248

If the replacement is planned, I think it would be better to use `LazyCell` than `once_cell::Lazy` in new code like this one, to not increase technical debt.

<img width="640" alt="image" src="https://github.com/user-attachments/assets/c90bdc76-f98c-4f96-a437-a0b48e09903e" />